atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -771,6 +792,20 @@ private void finishConsumingPreviousRow(PeekingIterator<Map.Entry<Cell, Value>>
         };
     }
 
+    private static RowColumnRangeIterator emptyRowColumnRangeIterator() {
+        return new RowColumnRangeIterator() {
+            @Override
+            public boolean hasNext() {
+                return false;
+            }
+
+            @Override
+            public Map.Entry<Cell, Value> next() {
+                throw new NoSuchElementException();
+            }
+        };
+    }
+
```

Comment:
```suggestion
    private enum EmptyRowColumnRangeIterator implements RowColumnRangeIterator {
        INSTANCE;

        @Override
        public boolean hasNext() {
            return false;
        }

        @Override
        public Map.Entry<Cell, Value> next() {
            throw new NoSuchElementException();
        }
    }

```

Reply:
yep (will merge in other PR) - makes sense